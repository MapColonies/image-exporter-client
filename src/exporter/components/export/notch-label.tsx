import React from 'react';

interface NotchLabelProps {
  text: string;
}

export const NotchLabel: React.FC<NotchLabelProps> = ({ text }) => {
  return (
    <span style={{ backgroundColor: 'white', padding: '0 4px 0 4px' }}>
      { text }
    </span>
  );
}
