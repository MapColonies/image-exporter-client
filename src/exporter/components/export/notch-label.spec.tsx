import React from 'react';
import { shallow } from 'enzyme';
import { NotchLabel } from './notch-label';

const text = 'Test label';

describe('NotchLabel component', () => {
  it('renders correctly', () => {

    const wrapper = shallow(
      <NotchLabel text={text} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('notch label renders as SPAN', () => {
    const wrapper = shallow(
      <NotchLabel text={text} />
    );

    const labelContainer = wrapper.find('span');
    expect(labelContainer.text()).toBe(text); 
  });

});
