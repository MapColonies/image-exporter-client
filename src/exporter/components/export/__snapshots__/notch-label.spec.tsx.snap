// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotchLabel component renders correctly 1`] = `
<span
  style={
    Object {
      "backgroundColor": "white",
      "padding": "0 4px 0 4px",
    }
  }
>
  Test label
</span>
`;
